nifi-nar-bundles/nifi-zendesk-bundle/nifi-zendesk-processors/src/test/java/org/apache/nifi/processors/zendesk/GetZendeskTest.java

```diff
@@ -0,0 +1,276 @@
+import org.apache.nifi.web.client.api.HttpUriBuilder;
+import org.apache.nifi.web.client.provider.service.StandardWebClientServiceProvider;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentest4j.AssertionFailedError;
+
+public class GetZendeskTest {
+
+    private static final int HTTP_OK = 200;
+    private static final int HTTP_BAD_REQUEST = 400;
+
+    private static final String DEFAULT_QUERY_START_TIMESTAMP = "1640995200";
+    private static final String DEFAULT_CURSOR_VALUE = "123456789";
+    private static final String EMPTY_RESPONSE = "{}";
+    private static final String RESPONSE_WITH_CURSOR_FIELD_TEMPLATE = "{ \"%s\": " + DEFAULT_CURSOR_VALUE + " }";
+    private static final String THREE_RECORDS = "[{\"id\":1},{\"id\":2},{\"id\":3}]";
+    private static final String RESPONSE_WITH_THREE_RECORDS_TEMPLATE = "{ \"%s\": " + THREE_RECORDS + " }";
+    private static final String RESPONSE_WITH_ZERO_RECORDS_TEMPLATE = "{ \"%s\": [] }";
+
+    private MockWebServer server;
+    private TestRunner testRunner;
+
+    private static Stream<Arguments> supportedZendeskResourcesExportMethodCombinations() {
+        return Stream.of(ZendeskResource.values())
+            .flatMap(zendeskResource -> zendeskResource.getSupportedExportMethods()
+                .stream()
+                .map(exportMethod -> Arguments.of(zendeskResource, exportMethod)));
+    }
+
+    private static Stream<Arguments> unsupportedZendeskResourcesExportMethodCombinations() {
+        return Stream.of(ZendeskResource.values())
+            .flatMap(zendeskResource -> {
+                List<ZendeskExportMethod> zendeskExportMethods = new ArrayList<>(asList(ZendeskExportMethod.values()));
+                zendeskExportMethods.removeAll(zendeskResource.getSupportedExportMethods());
+                return zendeskExportMethods.stream()
+                    .map(exportMethod -> Arguments.of(zendeskResource, exportMethod));
+            });
+    }
+
+    @BeforeEach
+    public void init() throws IOException, InitializationException {
+        server = new MockWebServer();
+        server.start();
+
+        testRunner = newTestRunner(new TestGetZendesk());
+
+        StandardWebClientServiceProvider standardWebClientServiceProvider = new StandardWebClientServiceProvider();
+        String standardWebClientServiceProviderId = "standardWebClientServiceProvider";
+        testRunner.addControllerService(standardWebClientServiceProviderId, standardWebClientServiceProvider);
+        testRunner.enableControllerService(standardWebClientServiceProvider);
+
+        testRunner.setProperty(WEB_CLIENT_SERVICE_PROVIDER_NAME, standardWebClientServiceProviderId);
+        testRunner.setProperty(ZENDESK_SUBDOMAIN_NAME, "zendesk-subdomain");
```

Comment:
Is it possible to use the constants defined in the main class?

Reply:
Maybe I'm missing something, but the property names constants are used from the main class. The property values are arbitrary string, not relevant for the result of the tests, need to be set only because those are required properties.

Reply:
My bad, just the name was the same as the constant and it threw me off. It is good as is.